cirq-google/cirq_google/json_test_data/InternalGate.json

```diff
@@ -0,0 +1,8 @@
+{
+    "cirq_type": "InternalGate",
+    "gate_name": "CouplerDelayZ",
+    "gate_module": "pyle.cirqtools.pyle_gates",
+    "num_qubits": 2,
+    "delay": 1,
```

Comment:
How will units be treated here? This should probably be something like `delay_ns` instead.

Reply:
@maffoo this just tests serialization. `units` are not part of cirq but are part of `labrad` so I can't create an instance of a value with unit here.

This is not an issue however. InternalGate doesn't create the gate it's just a place holder. The gate will be created by pyle. so if the value with units was provided. it will work.

I will add a test for this when the I create the pyle part.

Reply:
Do we not want to support serializing `InternalGate` instances that use units and sending them through quantum engine, or using them without access to internal code like values with units? If we want to support such access then we would need a way to specify units without actually creating a value object. But if we don't want to support such use cases then I'm not sure how useful it is to have `InternalGate` in cirq_google, since it could only be used with internal gates that have units when you have access to the internal code.

Reply:
the serialization will happen through protos. In a follow up PR I will introduce a proto for InternalGate and the serialization will happen using it. the values of the proto will be of type `Program.Arg`

Reply:
Right, but `Program.Arg` does not support units, and in order to instantiate a `TestDelayZ` gate when deserializing we will need to know the units to create a `Value` instance. We can communicate the units in the _keys_ of the `gate_args`, e.g. using `delay_ns` here and then having a mechanism when deserializing to convert that to `delay` with a value in `ns`; or else we need some other mechanism. Note that this doesn't directly impact the code you've written here, it just might mean this example would need to change. My main concern is to be sure we've thought about how we plan to deal with units, because we use them in lots of places for parameters on internal gates.

Reply:
`Program.Arg` lacks a couple of things not just units. it also lacks the ability to set a value as a sequence unless that sequence is of boolean values.. this for example doesn't allow us to create gates that take `qid_shape` as input.


I was intending to update `Program.Arg` to support sequence values since that was the first thing that I noticed. so thanks for bringing the units point because they are also needed. 

so the entire work will be in pyle to update the `Arg` proto.

Reply:
this should be resolved using the new InternalClass, PTAL